fdbclient/BlobCipher.cpp

```diff
@@ -618,9 +689,10 @@ EncryptBlobCipherAes265Ctr::~EncryptBlobCipherAes265Ctr() {
 
 DecryptBlobCipherAes256Ctr::DecryptBlobCipherAes256Ctr(Reference<BlobCipherKey> tCipherKey,
                                                        Reference<BlobCipherKey> hCipherKey,
-                                                       const uint8_t* iv)
+                                                       const uint8_t* iv,
+                                                       BlobCipherMetrics::UsageType usageType)
   : ctx(EVP_CIPHER_CTX_new()), textCipherKey(tCipherKey), headerCipherKey(hCipherKey),
-    headerAuthTokenValidationDone(false), authTokensValidationDone(false) {
+    headerAuthTokenValidationDone(false), authTokensValidationDone(false), usageType(usageType) {
 	if (ctx == nullptr) {
```

Comment:
please add `assert` ensuring validation of `type` input parameter.

Reply:
I'm adding the assert to `BlobCipherMetrics::counter()` instead.